content/openusercss-integration.js

```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+(() => {
+  const manifest = chrome.runtime.getManifest();
+
+  // Tell the page that we exist and that it should send the handshake
+  window.postMessage({
+    'type': 'ouc-begin-handshake'
+  }, '*');
+
+  // Wait for the handshake
+  window.addEventListener('message', event => {
+    if (event.data && event.data.type === 'ouc-handshake-question') {
```

Comment:
You should check `event.origin` too.

Reply:
The manifest already defines that the content script can only run on the site itself. Are there other cases in which an event could be received?

Reply:
See my comment above.